File: MiniDOMBase/DMD/Sistema/Notifiche/NotificaCursor.cs
using System;
using System.Collections;
using System.Collections.Generic;
using DMD;
using DMD.XML;
using DMD.Databases;
using DMD.Databases.Collections;
using minidom;
using minidom.repositories;
using static minidom.Sistema;
using static minidom.Anagrafica;

namespace minidom
{
    public partial class Sistema
    {


        /// <summary>
        /// Cursore sulla tabella delle notifiche
        /// </summary>
        /// <remarks></remarks>
        [Serializable]
        public class NotificaCursor 
            : minidom.Databases.DBObjectCursorPO<Notifica>
        {

            private DBCursorField<DateTime> m_Data = new DBCursorField<DateTime>("Data");
            //private DBCursorFieldDBDate m_Data = new DBCursorFieldDBDate("DataStr");
            private DBCursorStringField m_Context = new DBCursorStringField("Context");
            private DBCursorStringField m_SourceName = new DBCursorStringField("SourceName");
            private DBCursorField<int> m_SourceID = new DBCursorField<int>("SourceID");
            private DBCursorField<int> m_TargetID = new DBCursorField<int>("TargetID");
            private DBCursorStringField m_TargetName = new DBCursorStringField("TargetName");
            private DBCursorStringField m_Descrizione = new DBCursorStringField("Descrizione");
            private DBCursorField<DateTime> m_DataConsegna = new DBCursorField<DateTime>("DataConsegna");
            private DBCursorField<DateTime> m_DataLettura = new DBCursorField<DateTime>("DataLettura");
            private DBCursorField<StatoNotifica> m_StatoNotifica = new DBCursorField<StatoNotifica>("StatoNotifica");
            private DBCursorStringField m_Categoria = new DBCursorStringField("Categoria");

            /// <summary>
            /// Costruttore
            /// </summary>
            public NotificaCursor()
            {
            }


            /// <summary>
            /// Data
            /// </summary>
            public DBCursorField<DateTime> Data
            {
                get
                {
                    return this.m_Data;
                }
            }

            /// <summary>
            /// Categoria
            /// </summary>
            public DBCursorStringField Categoria
            {
                get
                {
                    return m_Categoria;
                }
            }

            /// <summary>
            /// Context
            /// </summary>
            public DBCursorStringField Context
            {
                get
                {
                    return m_Context;
                }
            }

            /// <summary>
            /// SourceName
            /// </summary>
            public DBCursorStringField SourceName
            {
                get
                {
                    return m_SourceName;
                }
            }

            /// <summary>
            /// SourceID
            /// </summary>
            public DBCursorField<int> SourceID
            {
                get
                {
                    return m_SourceID;
                }
            }

            /// <summary>
            /// TargetID
            /// </summary>
            public DBCursorField<int> TargetID
            {
                get
                {
                    return m_TargetID;
                }
            }

            /// <summary>
            /// TargetName
            /// </summary>
            public DBCursorStringField TargetName
            {
                get
                {
                    return m_TargetName;
                }
            }

            /// <summary>
            /// Descrizione
            /// </summary>
            public DBCursorStringField Descrizione
            {
                get
                {
                    return m_Descrizione;
                }
            }

            /// <summary>
            /// DataConsegna
            /// </summary>
            public DBCursorField<DateTime> DataConsegna
            {
                get
                {
                    return m_DataConsegna;
                }
            }

            /// <summary>
            /// DataLettura
            /// </summary>
            public DBCursorField<DateTime> DataLettura
            {
                get
                {
                    return m_DataLettura;
                }
            }

            /// <summary>
            /// StatoNotifica
            /// </summary>
            public DBCursorField<StatoNotifica> StatoNotifica
            {
                get
                {
                    return m_StatoNotifica;
                }
            }

            //protected internal override CDBConnection GetConnection()
            //{
            //    return Notifiche.Database;
            //}

            /// <summary>
            /// Repository
            /// </summary>
            /// <returns></returns>
            protected override CModulesClass GetModule()
            {
                return minidom.Sistema.Notifiche;
            }

            //public override string GetTableName()
            //{
            //    return "tbl_SYSNotify";
            //}
        }
    }
}

File: MiniDOMBase/DMD/Sistema/Notifiche/NotificaCursor.vb
Imports minidom.Databases

Partial Public Class Sistema


    ''' <summary>
    ''' Cursore sulla tabella delle notifiche 
    ''' </summary>
    ''' <remarks></remarks>
    Public Class NotificaCursor
        Inherits DBObjectCursorPO(Of Notifica)

        'Private m_Data As New CCursorField(Of Date)("Data")
        Private m_Data As New CCursorFieldDBDate("DataStr")
        Private m_Context As New CCursorFieldObj(Of String)("Context")
        Private m_SourceName As New CCursorFieldObj(Of String)("SourceName")
        Private m_SourceID As New CCursorField(Of Integer)("SourceID")
        Private m_TargetID As New CCursorField(Of Integer)("TargetID")
        Private m_TargetName As New CCursorFieldObj(Of String)("TargetName")
        Private m_Descrizione As New CCursorFieldObj(Of String)("Descrizione")
        Private m_DataConsegna As New CCursorField(Of Date)("DataConsegna")
        Private m_DataLettura As New CCursorField(Of Date)("DataLettura")
        Private m_StatoNotifica As New CCursorField(Of StatoNotifica)("StatoNotifica")
        Private m_Categoria As New CCursorFieldObj(Of String)("Categoria")

        Public Sub New()
        End Sub

        'Public ReadOnly Property Data As CCursorField(Of Date)
        '    Get
        '        Return Me.m_Data
        '    End Get
        'End Property

        Public ReadOnly Property Data As CCursorFieldDBDate
            Get
                Return Me.m_Data
            End Get
        End Property


        Public ReadOnly Property Categoria As CCursorFieldObj(Of String)
            Get
                Return Me.m_Categoria
            End Get
        End Property

        Public ReadOnly Property Context As CCursorFieldObj(Of String)
            Get
                Return Me.m_Context
            End Get
        End Property

        Public ReadOnly Property SourceName As CCursorFieldObj(Of String)
            Get
                Return Me.m_SourceName
            End Get
        End Property

        Public ReadOnly Property SourceID As CCursorField(Of Integer)
            Get
                Return Me.m_SourceID
            End Get
        End Property

        Public ReadOnly Property TargetID As CCursorField(Of Integer)
            Get
                Return Me.m_TargetID
            End Get
        End Property

        Public ReadOnly Property TargetName As CCursorFieldObj(Of String)
            Get
                Return Me.m_TargetName
            End Get
        End Property

        Public ReadOnly Property Descrizione As CCursorFieldObj(Of String)
            Get
                Return Me.m_Descrizione
            End Get
        End Property

        Public ReadOnly Property DataConsegna As CCursorField(Of Date)
            Get
                Return Me.m_DataConsegna
            End Get
        End Property

        Public ReadOnly Property DataLettura As CCursorField(Of Date)
            Get
                Return Me.m_DataLettura
            End Get
        End Property

        Public ReadOnly Property StatoNotifica As CCursorField(Of StatoNotifica)
            Get
                Return Me.m_StatoNotifica
            End Get
        End Property


        Protected Friend Overrides Function GetConnection() As Databases.CDBConnection
            Return Notifiche.Database
        End Function

        Protected Overrides Function GetModule() As CModule
            Return Nothing
        End Function

        Public Overrides Function GetTableName() As String
            Return "tbl_SYSNotify"
        End Function

    End Class


End Class